src/bit-systems/loop-animation.ts

```diff
@@ -0,0 +1,105 @@
+import { addComponent, defineQuery, enterQuery, exitQuery, hasComponent, removeComponent } from "bitecs";
+import { AnimationAction, AnimationClip, AnimationMixer, LoopRepeat } from "three";
+import {
+  MixerAnimatable,
+  MixerAnimatableData,
+  LoopAnimation,
+  LoopAnimationData,
+  LoopAnimationInitialize,
+  LoopAnimationInitializeData,
+  Object3DTag
+} from "../bit-components";
+import { HubsWorld } from "../app";
+
+const loopAnimationInitializeQuery = defineQuery([LoopAnimationInitialize, MixerAnimatable, Object3DTag]);
+const loopAnimationInitializeEnterQuery = enterQuery(loopAnimationInitializeQuery);
+
+const loopAnimationQuery = defineQuery([LoopAnimation, MixerAnimatable, Object3DTag]);
+const loopAnimationExitQuery = exitQuery(loopAnimationQuery);
+
+// Question: Who should have AnimationMixer?
+//           MixerAnimatable component or Scene/App?
+
+const getActiveClips = (
+  animations: Array<AnimationClip>,
+  activeClipIndices: number[],
+  clip: string
+): AnimationClip[] => {
+  if (clip !== "") {
+    const activeClips = [];
+    const clipNames = clip.split(",");
+    for (let i = 0; i < clipNames.length; i++) {
+      const clipName = clipNames[i];
+      const foundClip = animations.find((clip: AnimationClip) => {
+        return clip.name === clipName;
+      });
+      if (foundClip) {
+        activeClips.push(foundClip);
+      } else {
+        console.warn(`Could not find animation names '${clipName}' in `, animations);
+      }
+    }
+    return activeClips;
+  } else {
+    return activeClipIndices.map((index: number) => animations[index]);
+  }
+};
+
+export function loopAnimationSystem(world: HubsWorld): void {
+  loopAnimationInitializeEnterQuery(world).forEach((eid: number): void => {
+    const object = world.eid2obj.get(eid)!;
+    const mixer = MixerAnimatableData.get(eid)!;
+
+    addComponent(world, LoopAnimation, eid);
```

Comment:
I might be missing something since you ended up doing it twice. What does the additional `LoopAnimationInitialize` component get us vs doing this in an enter query for `LoopAnimation` itself?